Source/Engine/Core/Random.h

```diff
@@ -14,4 +14,92 @@ namespace Random
     {
         return (float)rand() / (float)RAND_MAX;
     }
+
+    /// <summary>
+    ///Generates a random number within a specified range.
+    /// </summary>
+    /// <param name="min">The minimum value of the range.</param>
+    /// <param name="max">The maximum value of the range.</param>
+    /// <returns>The random <see cref="float"/>.</returns>
+    inline float RandRange(float min, float max)
+    {
+        return min + (max - min) * Rand();
+    }
+
+    /// <summary>
+    /// Generates a random number within a specified range.
+    /// </summary>
+    /// <param name="min">The minimum value of the range.</param>
+    /// <param name="max">The maximum value of the range.</param>
+    /// <returns>The random <see cref="int"/>.</returns>
+    inline int RandRange(int min, int max)
+    {
+        return min + (max - min) * Rand();
```

Comment:
My remark here is that the C standard library `rand()` function isn't quite as nice as what C# has. (C's rand has so many weird edge cases that don't matter here.) And more importantly, they behave differently.

In C#, `random.Next(0,2)` will generate the values 0 and 1. [The maximum value is exclusive.](https://learn.microsoft.com/en-us/dotnet/api/system.random.next?view=net-7.0)
This also applies to the `Random.NextDouble` function in C#. `Random.NextDouble` will never generate a `1.0f`.

Meanwhile in C world, this class has 
```c
inline float Rand()
{
    return (float)rand() / (float)RAND_MAX;
}
```
Which absolutely can generate `1.0f`, even if the odds of it happening are really low.

This annoyingly leads to `return min + (max - min) * Rand();` having a super low chance of generating the `max` value. I can sketch up a diagram or a better explanation if you're curious.

Reply:
(I hope someday someone adds a proper random number generation library to Flax, just to put me out of my misery. :P)

Reply:
No worries. but I really was confused with OnSphere and Inside Sphere :D
Yeah, seems rand() is really the problem here. Maybe we would use <uniform_real_distribution> for it?
https://stackoverflow.com/questions/288739/generate-random-numbers-uniformly-over-an-entire-range

Seems Flax Engine really needs some custom random solution, like Unity does.

Reply:
I'm not a C++ expert, so I can't tell whether that's the best solution. It does seem reasonable though.

Feel free to split up this pull request into one that improves the C# side, and then leave improving the C++ for another pull request. After all, if we have to introduce a proper randomisation function or library to the C++ side, then that definitely warrants a bit more attention.

Reply:
I have reverted C++ Random.h changes. So now it`s only C# pull request

Reply:
I'm probably missing something, but shouldn't one of the C++ standard random generators suffice to replace C's `rand`? https://en.cppreference.com/w/cpp/numeric/random